arch/cpu/nrf52840/Makefile.nrf52840

```diff
@@ -23,13 +23,17 @@ endif
 ifeq ($(BOARD), dongle)
   LDSCRIPT ?= $(CONTIKI_CPU)/ld/nrf52840-bootloader.ld
   CFLAGS += -DBOARD_PCA10059
+  NRF52840_NATIVE_USB ?= 1
+  NRF52840_USB_DFU_TRIGGER ?= 1
 else
   LDSCRIPT ?= $(CONTIKI_CPU)/ld/nrf52840.ld
   CFLAGS += -DBOARD_PCA10056
+  NRF52840_NATIVE_USB ?= 0
+  override NRF52840_USB_DFU_TRIGGER = 0
```

Comment:
Suggest throwing a warning instead of overriding silently, like so

```
  ifeq ($(NRF52840_NATIVE_USB)$(NRF52840_USB_DFU_TRIGGER), 11)
    $(warning "Building for BOARD=dk")
    $(warning "with NRF52840_NATIVE_USB=1 NRF52840_USB_DFU_TRIGGER=1")
    $(warning "This is invalid. Overriding with NRF52840_USB_DFU_TRIGGER=0")
    override NRF52840_USB_DFU_TRIGGER = 0
  endif  
endif
```